### ISSUE TITLE
MPKitBranchMetrics crash when customerid identity is nil

### ISSUE THREAD

**Issue by saniul** [mParticle/mparticle-apple-sdk#79]
Crashes on [MPKitBranchMetrics.m#L187](https://github.com/mparticle-integrations/mparticle-apple-integration-branchmetrics/blob/5a5e55fb0d04f44eca5fdde9e73c33912fa599bf/mParticle-BranchMetrics/MPKitBranchMetrics.m#L187) because `identityString` is a `NSNull`, not an `NSString` in this case.

Exception:
```
-[NSNull length]: unrecognized selector sent to instance 0x10f945f08
```
Debugger backtrace: 
https://gist.github.com/saniul/2e7f4f3651b1333e0c1b2313b01aea27

Better backtrace in Xcode:
[Xcode backtrace screenshot](https://user-images.githubusercontent.com/371773/46095079-94874580-c1b3-11e8-9c44-15bab9039f2d.png)

Reproduction:
Ran into this when running the app in a simulator. User is not logged in so customerId and all identities except `other` are set to `nil` in the identity request provided when calling `.startWithOptions`. Can't really tell you any specific steps to get the device in a crashing state, sorry.

**Comment by samdozor** [comment]
@saniul we'll be able to push a fix to this to the Branch kit, and possible might make a larger change to core after that. 

I'm betting the steps to repro are something like, create an identity request like so:

```
MPIdentityApiRequest *request = [MPIdentityApiRequest requestWithEmptyUser];
[request setUserIdentity:nil identityType: MPUserIdentityCustomerId];
[request setUserIdentity:@"foo" identityType: MPUserIdentityOther]; 
```
And then pass that into either SDK init, login, or logout. The workaround for now would be to not set a nil user identity (just don't set the key at all):

```
MPIdentityApiRequest *request = [MPIdentityApiRequest requestWithEmptyUser];
[request setUserIdentity:@"foo" identityType: MPUserIdentityOther]; 
```

The reason why this is an issue at all is that we treat nil and NSNull as semantically meaning "delete this identity", so the SDK passes that through to kits which in this case isn't handling it appropriately.

**Comment by samdozor** [comment]
This has been [addressed here](https://github.com/mParticle/mparticle-apple-sdk/commit/d90a1eb769618bc109aa38f8a756f15c4120bd26) and released in 7.7.0.
